ui/app/news/[id]/page.tsx

```diff
@@ -1,20 +1,74 @@
-import NewsDetail from "../../../components/NewsDetail";
-import { VALIDATED_ENV } from "../../../lib/constants";
+"use client";
 
-async function getNewsData(id: string) {
-  const res = await fetch(`${VALIDATED_ENV.API_URL}/news/${id}`, { next: { revalidate: 60 } });
-  if (!res.ok) {
-    throw new Error("Failed to fetch news");
-  }
-  return res.json();
+import React from "react";
+import NewsDetail from "@/components/NewsDetail";
+import { useParams, useSearchParams } from "next/navigation";
+import { scroller } from "react-scroll";
+
+interface NewsType {
+  title: string;
+  sections: {
+    title: string;
+    content: string;
+    context: {
+      name: string;
+      url: string;
+      description: string;
+      provider: {
+        name: string;
+        image?: {
+          thumbnail: {
+            contentUrl: string;
+          };
+        };
+      }[];
+      datePublished: string;
+      image?: {
+        contentUrl: string;
+        thumbnail: {
+          contentUrl: string;
+          width: number;
+          height: number;
+        };
+      };
+      article?: string;
+      score?: number;
+    }[];
+  }[];
```

Comment:
Better to separate in smaller interface, so we can easily reuse and extend them later.